src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/MethodWriter.java

```diff
@@ -101,6 +101,8 @@ protected void buildMethodDoc(Content detailsList) {
         if (!methods.isEmpty()) {
             Content methodDetailsHeader = getMethodDetailsHeader(detailsList);
             Content memberList = writer.getMemberList();
+            writer.addToTableOfContents(HtmlIds.METHOD_DETAIL, contents.methodDetailLabel);
+            writer.tocBuilder.pushNested(HtmlTree.OL(HtmlStyle.tocList));
```

Comment:
Coding-style question:
Rather than have multiple members (methods and fields) on `HtmlDocletWriter` would it be better to treat the TOC  as a first class item for each page and put all the functionality there?  Or maybe that's a separate cleanup for later.